### ISSUE TITLE
task model: F bounded quantification bug

### ISSUE THREAD

**Issue by vsmenon** [dart-lang/dev_compiler#357]
From f_bounded_quantification2_test.dart:

```dart
class Entities<T extends ConceptEntity<T>> implements EntitiesApi<T> { }

class ConceptEntity<T extends ConceptEntity<T>> implements EntityApi { }

abstract class EntityApi<T extends EntityApi<T>> { }

abstract class EntitiesApi<T extends EntityApi<T>> { }

class Concept extends ConceptEntity<Concept> { }

main() {
  new ConceptEntity<Concept>();
  new ConceptEntity<ConceptEntity>();
}
```

is triggering the following errors - newly in the task model strong mode:

```
severe: [AnalyzerMessage] 'T' does not extend 'EntityApi<T>' (test/codegen/language/f_bounded_quantification2_test.dart, line 7, col 67)
severe: [AnalyzerMessage] 'ConceptEntity' does not extend 'ConceptEntity<ConceptEntity>' (test/codegen/language/f_bounded_quantification2_test.dart, line 19, col 21)
```

This particular test isn't strictly blocking - we don't support this pattern yet in codegen, but it is a regression.

**Comment by vsmenon** [comment]
We're now supporting codegen for this, but those errors are still there.

**Comment by jmesserly** [comment]
moved to https://github.com/dart-lang/sdk/issues/25530
